src/main/java/com/ericsson/ei/mongo/MongoDBHandler.java

```diff
@@ -363,42 +361,62 @@ private boolean doDrop(String dataBaseName, String collectionName, MongoQuery qu
+                createCollection(databaseName, collectionName);
+            }
+        } catch (MongoClientException e) {
+            LOGGER.error("Failure when handling Mongo collection: {} ", e.getMessage(), e);
+            return null;
+        }
+
+        MongoDatabase db = mongoClient.getDatabase(databaseName);
+        MongoCollection<Document> collection = db.getCollection(collectionName);
+        return collection;
+    }
+
+    private List<String> getCollectionList(String databaseName) {
+        MongoDatabase db = mongoClient.getDatabase(databaseName);
         List<String> collectionList;
 
         try {
-            db = mongoClient.getDatabase(dataBaseName);
             collectionList = db.listCollectionNames().into(new ArrayList<String>());
         } catch (MongoInterruptedException | MongoSocketReadException | MongoSocketWriteException
                 | MongoCommandException | IllegalStateException e) {
-            LOGGER.error("Failed to get Mongo collection, Reason : {} ", e.getMessage(), e);
+            String message = "Failed to get Mongo collection list, Reason: " + e.getMessage();
             closeMongoDbConnection();
-            return null;
+            throw new MongoClientException(message, e);
         }
 
-        if (!collectionList.contains(collectionName)) {
-            LOGGER.debug(
-                    "The requested database({}) / collection({}) not available in mongodb, Creating ........",
-                    dataBaseName, collectionName);
-            try {
-                db.createCollection(collectionName);
-            } catch (MongoCommandException e) {
-                String message = "collection '" + dataBaseName + "." + collectionName
-                        + "' already exists";
-                if (e.getMessage().contains(message)) {
-                    LOGGER.warn("A {}.", message, e);
-                } else {
-                    throw e;
-                }
+        return collectionList;
+    }
+
+    private void createCollection(String databaseName, String collectionName) {
+        MongoDatabase db = mongoClient.getDatabase(databaseName);
+        LOGGER.debug("The requested database({}) / collection({}) not found in mongodb, Creating.",
+                databaseName, collectionName);
+        try {
+            db.createCollection(collectionName);
+        } catch (MongoCommandException e) {
+            String collectionExistsError = String.format("collection '%s.%s' already exists",
+                    databaseName, collectionName);
+            if (e.getMessage().contains(collectionExistsError)) {
+                // When multiple operations try to create the same collection a collection may
+                // already exist.
+                LOGGER.warn("A {}.", collectionExistsError, e);
+            } else {
+                String message = "Failed to create Mongo collection, Reason: " + e.getMessage();
+                throw new MongoClientException(message, e);
```

Comment:
Try bringing the logic from the try/catch into their own methods.

Reply:
Done in next commit.